CRM/Utils/Geocode/Google.php

```diff
@@ -82,7 +82,7 @@ public static function format(&$values, $stateName = FALSE) {
       $add .= ',+';
     }
 
-    if (!empty($values['state_province']) || !empty($values['state_province_id'])) {
+    if (!empty($values['state_province']) || (!empty($values['state_province_id']) && $values['state_province_id'] != 'null')) {
```

Comment:
Thanks. IMO that's incorrect behaviour of the form then - it shouldn't be sending 'null' as a string, surely? Weirdness! The state province ID should conform to an integer or literal null, not the string ... unless there's a good reason to tolerate the other. /opinion